Assets/HotmaxEditorVR/activeProp.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class activeProp : MonoBehaviour
+{
+
+    //--local refs
+    private GameObject _selectedObject = stateManager.selectedObject;
+    private bool _selectedObjectIsActive = stateManager.selectedObjectIsActive;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+    //--put the monobehaviours into a list
+    private List<MonoBehaviour> decorators = new List<MonoBehaviour>();
+
+    private void OnEnable()
+    {
+        stateManager.selectedObjectEvent += updateSelectedObject;
+        stateManager.editorModeEvent += updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent += updateSelectedObjectIsActive;
+        addDecorations();
+    }
+
+    private void OnDisable()
+    {
+        stateManager.selectedObjectEvent -= updateSelectedObject;
+        stateManager.editorModeEvent -= updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent -= updateSelectedObjectIsActive;
+        removeDecorators();
+    }
+
+    //this only fires when a new object is selected
+    void updateSelectedObject(GameObject value)
+    {
+        _selectedObject = value;
+        Destroy(this);
+    }
+
+    private void Awake()
+    {
+        //--This is a check for if the instantiated object is a clone
+        if (this.gameObject != _selectedObject)
+        {
+            Destroy(this);
+        }
+    }
+
+
+    // -- ENTRY POINT : this object is selected
```

Comment:
change `ENTRY POINT` to `RUN STATE MACHINE` + and down below as well

Reply:
DONE